queries/python/highlights.scm

```diff
@@ -276,11 +276,27 @@
  (#any-of? @constructor "__new__" "__init__"))
 
 ; First parameter of a method is self or cls.
+((class_definition
+  body: (block
+    (decorated_definition
+      (decorator (identifier) @_classmethod)
+        definition: (function_definition
+          parameters: (parameters . (identifier) @variable.builtin)))))
+ (#eq? @variable.builtin "cls")
+ (#eq? @_classmethod "classmethod"))
+((class_definition
+  body: (block
+    (decorated_definition
+      (decorator (identifier) @_not_staticmethod)
+        definition: (function_definition
+          parameters: (parameters . (identifier) @variable.builtin)))))
+ (#eq? @variable.builtin "self")
+ (#not-eq? @_not_staticmethod "staticmethod"))
```

Comment:
Is this really necessary? Are there people who use the variable name self in staticmethods?

Reply:
Really necessary? No. I mean, self in python is just a convention anyway. But self as an implicit argument to the method does not exist for `@staticmethod` and if 

https://github.com/nvim-treesitter/nvim-treesitter/blob/b48f9b039ba2b52aa9677012f3882056e83202b0/queries/python/highlights.scm#L135-L136

would not exist, this would be reflected by the highlight. 

But I'm fine removing that part from the query to reduce maintenance burden.

Reply:
No, it's ok. I just wanted to ask about your motivation